src/main/java/net/obvj/performetrics/Counter.java

```diff
@@ -299,9 +301,10 @@ public void setUnitsAfter()
      *         units are set; or the difference between {@code unitsBefore} and the current
      *         value retrieved by the counter's time source, if {@code unitsAfter} is not set
      */
-    protected long elapsedTimeInternal()
+    long elapsedTimeInternal()
     {
-        long tempUnitsAfter = unitsAfterFlag ? unitsAfter : type.getTime(timeUnit);
+        if (!unitsBeforeSet) return 0;

```

Comment:
Remove this to preserve natural behavior. Instead, evaluate mocking the system call to get the new elapsed time.